src/weather.cpp

```diff
@@ -559,6 +502,56 @@ void weather_effect::acid()
     }
 }
 
+double precip_mm_per_hour( precip_class const p )
+// Precipitation rate expressed as the rainfall equivalent if all
+// the precipitation were rain (rather than snow).
+{
+    return
+        p == precip_class::VERY_LIGHT ? 0.5 :
+        p == precip_class::LIGHT ? 1.5 :
+        p == precip_class::HEAVY ? 3   :
+        0;
+}
+
+void handle_weather_effects( weather_type const w )
+{
+    if( weather::rains( w ) && weather::precip( w ) != precip_class::NONE ) {
+        fill_water_collectors( precip_mm_per_hour( weather::precip( w ) ), weather::acidic( w ) );
+        int wetness = 0;
+        time_duration decay_time = 60_turns;
+        if( weather::precip( w ) == precip_class::VERY_LIGHT ) {
+            wetness = 5;
+            decay_time = 5_turns;
+        } else if( weather::precip( w ) == precip_class::LIGHT ) {
+            wetness = 30;
+            decay_time = 15_turns;
+        } else if( weather::precip( w ) == precip_class::HEAVY ) {
+            decay_time = 45_turns;
+            wetness = 60;
+        }
+        g->m.decay_fields_and_scent( decay_time );
+        weather_effect::wet_player( wetness );
+    }
+    glare( w );
+    std::vector<std::pair<std::string, int>> weather_effects = weather::effects( w );
+
+    std::map<std::string, weather_effect_fn> all_weather_effects = {
```

Comment:
This conversion (string from JSON --> C++ function pointer) should be done when the weather types are loaded from JSON. If there is any problem with those strings, you want to inform the player (or the modder) as early as possible.

It would also save quite some processing time during the game.